### PR TITLE
Move column sorting properties to a column mixin

### PR DESCRIPTION
**Pull request by mollykreis**
# Pull Request

## 🤨 Rationale

Currently, there is no way for a table column to not support sorting because the sorting attributes (`sorting-disabled`, `sort-index`, and `sort-direction`) are defined on the base `TableColumn` class. The menu button column, however, should not allow sorting. Therefore, the three sorting attributes are being moved off of the base `TableColumn` class and into a mixin, that every existing column now uses.

## 👩‍💻 Implementation

**nimble-components**
- Create mixin for sortable columns
- Remove sorting properties from the base `TableColumn` class
- Add `sortingDisabled` to column internals

**nimble-angular**
- Remove sorting properties from the base column directive and into each column type directive
- Add support for `sortingDisabled` (including tests) on all columns since this hadn't been exposed through Angular

**nimble-blazor**
- Create `ISortableColumn` interface
- Move sorting properties from the base column component and into `ISortableColumn`
- Update all columns to implement `ISortableColumn`
- Add support for `SortingDisabled` since this hadn't been exposed through Blazor
- Update tests for sortable columns
- Update tests for base columns since it previously was only testing sorting properties. I've removed those and added tests for missing base properties of `ColumnId`, `ActionMenuSlot`, `ActionMenuLabel`, and `ColumnHidden`.

## 🧪 Testing

- Unit tests

## ✅ Checklist

- [ ] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Comment by mollykreis** [comment]
@m-akinc, will you buddy this PR for me?

File: packages/nimble-components/src/table-column/text-base/index.ts [hunk 1]
```diff
@@ -36,6 +37,8 @@ export function mixinTextBase<
     TColumnValidator extends ColumnValidator<[]> = ColumnValidator<[]>
 >(base: TBase) {
     return mixinGroupableColumnAPI(
-        mixinFractionalWidthColumnAPI(mixinColumnWithPlaceholderAPI(base))
+        mixinFractionalWidthColumnAPI(
+            mixinColumnWithPlaceholderAPI(mixinSortableColumnAPI(base))
+        )
```
Comment (m-akinc) [inline packages/nimble-components/src/table-column/text-base/index.ts:42]: nit
```suggestion
        mixinFractionalWidthColumnAPI(
            mixinColumnWithPlaceholderAPI(
                mixinSortableColumnAPI(base)
            )
        )
```

**Review by m-akinc — APPROVED** [review]

File: packages/nimble-components/src/table-column/text-base/index.ts (hunk 1 above)
Comment (mollykreis) [reply packages/nimble-components/src/table-column/text-base/index.ts:42]: Prettier does not agree with your suggestion 😄 

Assuming you don't feel really strongly about this specific formatting, I think I'll let prettier have its way here.

Comment (m-akinc) [reply packages/nimble-components/src/table-column/text-base/index.ts:42]: Sometimes I hate prettier.

File: packages/nimble-components/src/table-column/mixins/sortable-column.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    TBase extends SortableTableColumnConstructor
+>(base: TBase) {
+    /**
+     * The Mixin that provides a concrete column with the API to allow sorting
+     * by the values in that column.
+     */
+    abstract class SortableColumn extends base {
+        public sortingDisabled = false;
+
+        public sortIndex?: number | null = null;
```
Comment (rajsite) [inline packages/nimble-components/src/table-column/mixins/sortable-column.ts:23]: Can we leave this uninitialized to undefined like it was previously in base? (prefer undefined over null)

```suggestion
        public sortIndex?: number | null;
```

Comment (rajsite) [reply packages/nimble-components/src/table-column/mixins/sortable-column.ts:23]: Hopefully we can do the same for groupIndex above as well

File: packages/nimble-components/src/table-column/mixins/groupable-column.ts [hunk 3]
```diff
@@ -26,11 +26,7 @@ export function mixinGroupableColumnAPI<
         }
 
         public groupIndexChanged(): void {
-            if (typeof this.groupIndex === 'number') {
-                this.columnInternals.groupIndex = this.groupIndex;
-            } else {
-                this.columnInternals.groupIndex = undefined;
-            }
+            this.columnInternals.groupIndex = this.groupIndex ?? undefined;
```
Comment (rajsite) [inline packages/nimble-components/src/table-column/mixins/groupable-column.ts:29]: Can you do the same in the fractional-width-column mixins for fractionalWidthChanged and minPixelWidthChanged?

File: packages/nimble-components/src/table-column/mixins/tests/sortable-column.spec.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            delegatedEvents: [],
+            validator: new ColumnValidator<[]>([])
+        };
+    }
+}
+
+// prettier-ignore
+async function setup(): Promise<Fixture<TestTableColumn>> {
+    return fixture(columnName);
+}
```
Comment (rajsite) [inline packages/nimble-components/src/table-column/mixins/tests/sortable-column.spec.ts:36]: no action needed (unless the ignore was accidental): prettier-ignore only impacts the following "node" right, i.e. just this function.. how badly does it mess this function up lol?

File: packages/nimble-components/src/table/models/table-update-tracker.ts [hunk 5]
```diff
@@ -118,9 +118,9 @@ export class TableUpdateTracker<
         ) {
             this.track('columnDefinition');
         } else if (
-            isColumnProperty(changedColumnProperty, 'sortingDisabled')
```
Comment (rajsite) [inline packages/nimble-components/src/table/models/table-update-tracker.ts:121]: no action needed (maybe a tech debt if it seems like a good direction): huh... half-thought but I think I like the idea of moving towards the table only relying on columnInternals and not the public api of columns. Seems like we are getting close.

File: change/@ni-nimble-angular-471f2c22-eef2-4640-9335-5fc12cb49c02.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (rajsite) [inline change/@ni-nimble-angular-471f2c22-eef2-4640-9335-5fc12cb49c02.json:1]: Great job churning through this so quick. Wish there wasn't so much boilerplate needed in the wrappers 😫 but looks good 🚢

**Review by rajsite — APPROVED** [review]

File: packages/nimble-components/src/table-column/mixins/tests/sortable-column.spec.ts (hunk 4 above)
Comment (mollykreis) [reply packages/nimble-components/src/table-column/mixins/tests/sortable-column.spec.ts:36]: I had copied this from the other mixin tests, but it doesn't look like it's actually needed. I'll remove it from all of them.

File: packages/nimble-components/src/table/models/table-update-tracker.ts (hunk 5 above)
Comment (mollykreis) [reply packages/nimble-components/src/table/models/table-update-tracker.ts:121]: I think we are getting close. I don't have a strong opinion at this point.

Comment (rajsite) [reply packages/nimble-components/src/table/models/table-update-tracker.ts:121]: Feels nice academically, but I can't think of a concrete reason to do it yet, so no action needed if no strong opinions 👍

**Comment by rajsite** [comment]
Gonna override merge since it's pretty mechanical and following existing patterns. If concerns @jattasNI @atmgrifter00 @msmithNI I'm sure we can address in a follow-up

**Pull request merged**